### ISSUE TITLE
Sourmash `load_file_as_signatures` on empty signatures produces IndexError

### ISSUE THREAD

**Issue by taylorreiter** [sourmash-bio/sourmash#1537]
Relevant to #1430 and being addressed in #1469. Running in sourmash 4.1.0. 

Using code:
```
    try:
        siglist = list(sourmash.load_file_as_signatures(args.sig_matches))
    except sourmash.exceptions.SourmashError:
        siglist = []
```

Produces error when ran on an empty signature:
```
Traceback (most recent call last):
  File "/home/tereiter/github/charcoal/.snakemake/conda/8763af208a097216f72fb7c92a9f00f6/lib/python3.9/runpy.py", line 197, in
 _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/tereiter/github/charcoal/.snakemake/conda/8763af208a097216f72fb7c92a9f00f6/lib/python3.9/runpy.py", line 87, in
_run_code
    exec(code, run_globals)
  File "/home/tereiter/github/charcoal/charcoal/contigs_search_taxonomy.py", line 130, in <module>
    returncode = cmdline(sys.argv[1:])
  File "/home/tereiter/github/charcoal/charcoal/contigs_search_taxonomy.py", line 125, in cmdline
    return main(args)
  File "/home/tereiter/github/charcoal/charcoal/contigs_search_taxonomy.py", line 37, in main
    siglist = list(sourmash.load_file_as_signatures(args.matches_sig))
  File "/home/tereiter/github/charcoal/.snakemake/conda/8763af208a097216f72fb7c92a9f00f6/lib/python3.9/site-packages/sourmash/
sourmash_args.py", line 381, in load_file_as_signatures
    db = _load_database(filename, yield_all_files)
  File "/home/tereiter/github/charcoal/.snakemake/conda/8763af208a097216f72fb7c92a9f00f6/lib/python3.9/site-packages/sourmash/
sourmash_args.py", line 304, in _load_database
    db = load_fn(filename,
  File "/home/tereiter/github/charcoal/.snakemake/conda/8763af208a097216f72fb7c92a9f00f6/lib/python3.9/site-packages/sourmash/
sourmash_args.py", line 227, in _multiindex_load_from_pathlist
    db = MultiIndex.load_from_pathlist(filename)
  File "/home/tereiter/github/charcoal/.snakemake/conda/8763af208a097216f72fb7c92a9f00f6/lib/python3.9/site-packages/sourmash$index.py", line 735, in load_from_pathlist
    file_list = load_pathlist_from_file(filename)
  File "/home/tereiter/github/charcoal/.snakemake/conda/8763af208a097216f72fb7c92a9f00f6/lib/python3.9/site-packages/sourmash$sourmash_args.py", line 397, in load_pathlist_from_file
    if not os.path.exists(file_list[0]):
IndexError: list index out of range
```

The error is triggered by this function, by this line of code `if not os.path.exists(file_list[0]):`.
The ValueError is never reached. 

```

def load_pathlist_from_file(filename):
    "Load a list-of-files text file."
    try:
        with open(filename, 'rt') as fp:
            file_list = [ x.rstrip('\r\n') for x in fp ]

        if not os.path.exists(file_list[0]):
            raise ValueError("first element of list-of-files does not exist")
    except OSError:
        raise ValueError(f"cannot open file '{filename}'")
    except UnicodeDecodeError:
        raise ValueError(f"cannot parse file '{filename}' as list of filenames")

    return file_list
```

My workaround for now is:

```
    try:
        siglist = list(sourmash.load_file_as_signatures(args.sig_matches))
    except IndexError:
        siglist = []
```

But it feels like a bad idea with unexpected edge cases to except all IndexErrors.

**Comment by keyabarve** [comment]
I'm taking care of this issue in #1469 :)

**Comment by ctb** [comment]
fixed by #1469 🎉
